src/models/tag/stream.rs

```diff
@@ -117,104 +184,66 @@ impl HotTags {
         Ok(())
     }
 
-    pub async fn get_global_tags_stream(
-        skip: Option<usize>,
-        limit: Option<usize>,
-        taggers_limit: Option<usize>,
-    ) -> Result<Option<Self>, DynError> {
-        let hot_tags = Self::try_from_index_sorted_set(
-            &TAG_GLOBAL_HOT,
-            None,
-            None,
-            skip,
-            limit,
-            SortOrder::Descending,
-        )
-        .await?
-        .unwrap_or_default();
-
-        if hot_tags.is_empty() {
-            return Ok(None);
+    fn get_cache_key_parts(tags_query: &HotTagsInput) -> Vec<String> {
+        match &tags_query.tagged_type {
+            Some(tagged) => vec![
+                tags_query.timeframe.to_string(),
+                tagged.to_string(),
+                tags_query.limit.to_string(),
+                tags_query.skip.to_string(),
```

Comment:
It seems unreasonable to have different caches for each `skip` and `limit` . In addition, this will make pagination buggy, as queries and expirations across "pages" might have a computed hot-tags at different times.